### ISSUE TITLE
JSON-LD to Turtle serialization eats up @id with no value

### ISSUE THREAD

**Issue by csarven** [solid/node-solid-server#497]
re: https://github.com/solid/node-solid-server/pull/496#issuecomment-286242305

`http://example.org/resourceA` contains:
`{ "@id": "http://example.net/note#foo", "http://schema.org/citation": { "@id": "http://example.org/article#results" } }`

```
$ curl -H'Accept: text/turtle' http://example.org/resourceA
@prefix note: <http://example.net/note#>.
@prefix schem: <http://schema.org/>.
@prefix n0: <http://example.org/article#>.

   note:foo schem:citation n0:results .
```

Looks good to me.

`http://example.org/resourceB` contains:
`{ "@id": "", "http://schema.org/citation": { "@id": "http://example.org/article#results" } }`

```
$ curl -H'Accept: text/turtle' http://example.org/resourceB

```

No body.

**Comment by dmitrizagidulin** [comment]
I don't think `"@id": ""` is valid JSON-LD. Try it in the [Distiller](http://rdf.greggkellogg.net/distiller)

**Comment by dmitrizagidulin** [comment]
Never mind, I think I see what you're saying. There's a problem with translation code in the `get` handler. (The base uri is somehow getting lost for translation)

**Comment by dmitrizagidulin** [comment]
Filed upstream issue https://github.com/linkeddata/rdflib.js/issues/177

**Comment by RubenVerborgh** [comment]
This is an rdflib bug related to JSON-LD and base IRIs, which is fixed in https://github.com/linkeddata/rdflib.js/pull/195. I'll wait for a new rdflib.js version to be released, and will then update package.json to fix this.

**Comment by RubenVerborgh** [comment]
Fixed by #548.
